sky/resources.py

```diff
@@ -986,6 +994,25 @@ def _try_validate_labels(self) -> None:
                     'The following labels are invalid:'
                     '\n\t' + invalid_table.get_string().replace('\n', '\n\t'))
 
+    def _try_validate_az_ml(self) -> None:
+        if not self.use_az_ml:
+            return
+        if self.ports is not None:
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(
+                    'Open ports are not supported for Azure Machine Learning.')
+        if (self.disk_tier is not None or
+                self.disk_size != _DEFAULT_DISK_SIZE_GB):
+            # Azure ML does not support custom disk size and disk tier.
+            # Reference: https://stackoverflow.com/questions/66923216/change-disk-type-azure-ml  # pylint: disable=line-too-long
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError('Custom disk size and disk tier are not '
+                                 'supported for Azure Machine Learning.')
+        if self.image_id is not None and self.extract_docker_image() is None:
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(
+                    'Custom image is not supported for Azure Machine Learning.')
+
```

Comment:
This is not clean. Why can't we add the check to unsupported_features of a specific cloud?